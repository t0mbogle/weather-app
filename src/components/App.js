import React, { useEffect, useState } from "react";
import "../styles/App.css";
import ForecastSummaries from "./ForecastSummaries";
import LocationDetails from "./LocationDetails";
import ForecastDetails from "./ForecastDetails";
import getForecast from "../requests/getForecast";

// Change this to const ES6 syntax after other changes
function App() {
  const [forecasts, setForecasts] = useState([]);
  const [location, setLocation] = useState({ city: "", country: "" });
  const [selectedDate, setSelectedDate] = useState(0); // App no longer recieves forecasts so 0 is the new default value

  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate
  ); // Searches through the array of forecasts and returns the matching value

  const handleForecastSelect = (date) => {
    setSelectedDate(date);
  };

  useEffect(() => {
    getForecast(setSelectedDate, setForecasts, setLocation);
  }, []); // Importing axios HTTP request from requests/getForecast

  return (
    <div className="weather-app">
      <h1>Weather App</h1>
      <LocationDetails city={location.city} country={location.country} />
      <ForecastSummaries
        forecasts={forecasts}
        onForecastSelect={handleForecastSelect}
        // This now gives ForecastSummaries component access to the prop 'onForecastSelect'
      />
      {selectedForecast && (
        <ForecastDetails
          forecast={selectedForecast} /* Conditional rendering */
        />
      )}
    </div>
  );
}

export default App;
