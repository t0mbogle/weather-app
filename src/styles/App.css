.weather-app {
    margin: 0 35px;
}

.titles-section {
    display: flex;
    flex-direction: row-reverse;
    justify-content: space-between;
    margin-bottom: 20px;
    font-size: 21px;
    color: white;
}