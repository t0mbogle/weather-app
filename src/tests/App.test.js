import { render, screen } from "@testing-library/react";
import App from "../components/App";
import forecast from "../data/forcast.json";

describe("App", () => {
  it("renders App component correctly", () => {
    render(<App location={forecast.location} />);
    const h1Element = screen.getByText(/Manchester, UK/i);

    expect(h1Element).toBeInTheDocument();
  });
});

// Make sure to update this test file whenever App gets updated
