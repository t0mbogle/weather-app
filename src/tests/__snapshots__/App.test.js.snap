// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <h1>
      Weather App
    </h1>
    <h1>
      , 
    </h1>
    <form
      class="search-form"
    >
      <input
        placeholder="Search UK city"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Search
      </button>
    </form>
    <div
      class="forecast-summaries"
    />
  </div>
</DocumentFragment>
`;
